crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -287,16 +300,80 @@ impl<'a> IRGenerator<'a> {
             obj_type,
             name: var_name,
             root: None,
-            def: Some(def),
+            def,
             witness,
             parent_block: self.context.current_block,
         };
         let v_id = self.context.add_variable(new_var, None);
         let v_value = Value::Single(v_id);
-        self.variable_values.insert(def, v_value);
+        if let Some(def) = def {
+            self.variable_values.insert(def, v_value);
+        }
         v_id
     }
 
+    //Helper function for create_new_value()
+    fn insert_new_struct(
+        &mut self,
+        def: Option<DefinitionId>,
+        values: Vec<(String, Value)>,
+    ) -> Value {
+        let result = Value::Struct(values);
+        if let Some(def_id) = def {
+            self.variable_values.insert(def_id, result.clone());
+        }
+        result
+    }
+
+    pub fn create_new_value(
+        &mut self,
+        typ: &noirc_frontend::Type,
+        base_name: &str,
+        def: Option<DefinitionId>,
+    ) -> Value {
+        match typ {
+            noirc_frontend::Type::Struct(_, t) => {
+                let mut values = Vec::new();
+                for i in &t.borrow().fields {
+                    let name = format!("{}.{}", base_name, i.0 .0.contents);
+                    let val = self.create_new_value(&i.1, &name, None);
+                    values.push((i.0 .0.contents.clone(), val));
```

Comment:
Minor style nitpick but I'd prefer `vecmap` over creating a vec manually and pushing to it in a loop. It conveys the meaning of mapping each element to a new corresponding Vec better and is slightly more efficient (from reserving the correct capacity in advance).
Not blocking over this though since it is minor.

Reply:
Done